fenix/app/src/main/java/org/mozilla/fenix/debugsettings/navigation/DebugDrawerDestination.kt

```diff
@@ -13,9 +13,11 @@ import androidx.compose.runtime.Composable
  * @property route The navigation route of the destination.
  * @property title The string ID of the destination's title.
  * @property content The destination's [Composable].
+ * @property onClick Invoked when the destination is clicked to be navigated to.
  */
 data class DebugDrawerDestination(
     val route: String,
     @StringRes val title: Int,
     val content: @Composable () -> Unit,
+    val onClick: () -> Unit,
```

Comment:
We'll want the @Composable lambda as our last parameter if we were to follow https://android.googlesource.com/platform/frameworks/support/+/androidx-main/compose/docs/compose-component-api-guidelines.md#parameters-order

Reply:
Maybe I am wrong here about this since the `content` isn't optional upon a second reading. Now I am wondering if we should prefer @Composable lambda before or after the callback parameters.

Reply:
I don't have a super strong opinion, and the current code won't even leverage it, but I'll update it for the QOL of any future consumers.